cluster.go

```diff
@@ -664,6 +675,37 @@ func (c *clusterClient) doresultfn(
 			} else {
 				nc = c.redirectOrNew(addr, cc, cm.Slot(), mode)
 			}
+			if hasInit && ei < i { // find out if there is a transaction block or not.
+				for mi = i; mi >= 0 && !isMulti(commands[mi]) && !isExec(commands[mi]); mi-- {
+				}
+				for ei = i; ei < len(commands) && !isMulti(commands[ei]) && !isExec(commands[ei]); ei++ {
+				}
```

Comment:
Both `mi` and `ei` cursor will stop at either `MULTI` or `EXEC` to avoid crossing tx boundaries.